### PR TITLE
Added New Climograph Dialogue On Climatic> Describe Menu 

### LINKED ISSUE #8657: Add climographs to the climatic menu
@jkmusyoka  Here is the Climatic > Describe menu.  It is very poor.

image

If you ask for climatic graphs it suggests bar charts and climographs are popular.  The recent dodoma summaries provide data with just monthly summaries of rainfall and temperatures.  I was not able to get a climaograph from the Describe > Graphs > General dialog - perhaps you can?

In addition could we have various attractive climographs perhaps using R code you suggest, but I would like to see what chatGPT could come up with?  I assume that will be built on code and hope it could suggest several different alternatives?

Then could we quickly construct an impressive dialog that implements these alternatives.

The basic structure is clear.

a) Usual data selector
b) Single receiver with label Rainfall.
c) Multiple receiver with label Temperature(s)  (It could just have mean monthly temperatures or Mean min and mean max, and perhaps other possibilities including extremes as well.
d) I assume there might be a second axis label to allow Multiplier and Offset for the secondary axis?
e) Be able to save the graph of course.
f) Other options dependin g on what there are.  For example maybe the max and min could have bands? Could there be an Alphs control to help the visibility of the different components?

I am keen of checking the sort of AI process as well as the dialog.

And I wonder about a Walter-Leith graph, from climatol, on another button on the same tab.  I think it can use the same data?

I would also like to buuld that Climatic > Describe menu more.

### PR DESCRIPTION
**Pull request by MeSophie**
Fixes #8657
@rdstern, @lilyclements  this PR replace PR #8846. Please have a look.

@lilyclements Please from a) to c) on @rdstern  comment can be remove from the Stand alone function?

### CODE REVIEW

**Comment by lilyclements** [comment]
@MeSophie apologies if starting a new PR was due to changes from me in the R code. I can outline the changes going forward on the original issue if you’d like? I didn’t notice comments a-c so thank you for drawing my attention to that. I’ll get to that today.

**Comment by MeSophie** [comment]
@lilyclements  No problem you can Still do it here. I think they are the only changes remaining from roger's comment.

**Review by rdstern — COMMENTED** [review]
@MeSophie and @lilyclements the dialog looks great and the facetting options now also work:  

image

@lilyclements with the facets there is still now the problems with the summaries and the overall title.  I wonder if those are just deleted when there are facets?

@MeSophie the one small problem is now with the save, where the control needs to move down a bit.  Perhaps also make the default title wl_graph?

image

**Comment by MeSophie** [comment]
@rdstern I improved the save control. Please have a look.

**Review by rdstern — COMMENTED** [review]
@MeSophie let's wait before merging to see whether @lilyclements will want to do the small changes in the plot, when there are multiple facets.
The code has adapted a function from the climaemet package.  Could you check that the comments in the current function give credit to that package as providing the source.

I tried the Describe > Use Graph dialog and report that this gave a slightly odd version of the graph?  I get the right version when using Prepare > R-Objects.  

image

**Comment by lilyclements** [comment]
@rdstern @MeSophie I have now addressed points a-c on the titles/subtitles/tags in the plot.

**Review by rdstern — APPROVED** [review]
@lilyclements and @MeSophie great.  That is an excellent addition to R-Instat!  @N-thony over to you to check and hopefully merge.

File: instat/frmMain.Designer.vb [hunk 1]
```diff
@@ -5112,9 +5111,7 @@ Partial Class frmMain
         Me.IsMdiContainer = True
         Me.MainMenuStrip = Me.mnuBar
         Me.Name = "frmMain"
-        Me.Text = "R-Instat " + My.Application.Info.Version.Major.ToString + "." +
-                My.Application.Info.Version.Minor.ToString + "." +
-                My.Application.Info.Version.Build.ToString
```
[the remark this answers is not in the document]
Comment (MeSophie) [reply instat/frmMain.Designer.vb:5125]: This Is solved on PR #8899.

File: instat/dlgClimograph.vb [hunk 2]
```diff
@@ -1,3 +1,363 @@
...
+    End Sub
+
+    Private Sub TestOKEnabled()
+        If rdoWalterLieth.Checked Then
+            If (Not ucrReceiverAbsolute.IsEmpty AndAlso Not ucrReceiverMaxtem.IsEmpty AndAlso Not ucrReceiverMintemp.IsEmpty AndAlso Not ucrReceiverMonth.IsEmpty AndAlso Not ucrReceiverRain.IsEmpty) OrElse Not ucrSave.IsComplete Then
+                ucrBase.OKEnabled(True)
+            Else
+                ucrBase.OKEnabled(False)
+            End If
+        End If
```
[the remark this answers is not in the document]
Comment (MeSophie) [reply instat/dlgClimograph.vb:186]: This Is solved on PR #8899.

File: instat/dlgClimograph.vb [hunk 3]
```diff
@@ -1,3 +1,363 @@
...
+        If bWrap Then
+            clsFacetFunction.SetRCommand("facet_wrap")
+        End If
+        If bRow OrElse bCol Then
+            clsFacetFunction.SetRCommand("facet_grid")
+        End If
+        If bRow Then
+            clsFacetOperator.AddParameter("left", clsROperatorParameter:=clsFacetRowOp, iPosition:=0)
+        ElseIf bCol AndAlso bWrap = False Then
```
[the remark this answers is not in the document]
Comment (MeSophie) [reply instat/dlgClimograph.vb:273]: This Is solved on PR #8899.

File: instat/dlgClimograph.vb [hunk 4]
```diff
@@ -1,3 +1,363 @@
...
+            clsFacetOperator.AddParameter("left", ".", iPosition:=0)
+        Else
+            clsFacetOperator.RemoveParameterByName("left")
+        End If
+        If bCol Then
+            clsFacetOperator.AddParameter("right", clsROperatorParameter:=clsFacetColOp, iPosition:=1)
+        ElseIf bRow AndAlso bWrap = False Then
```
[the remark this answers is not in the document]
Comment (MeSophie) [reply instat/dlgClimograph.vb:280]: This Is solved on PR #8899.

File: instat/dlgClimograph.vb [hunk 5]
```diff
@@ -1,3 +1,363 @@
...
+
+        If ucrSelectorClimograph.CurrentReceiver IsNot Nothing Then
+            ucrCurrentReceiver = ucrSelectorClimograph.CurrentReceiver
+        End If
+        ucr1stFactorReceiver.AddItemsWithMetadataProperty(ucrSelectorClimograph.ucrAvailableDataFrames.cboAvailableDataFrames.Text, "Climatic_Type", {"station_label"})
+        If ucrCurrentReceiver IsNot Nothing Then
+            ucrCurrentReceiver.SetMeAsReceiver()
+        End If
+        AddRemoveGroupBy()
+    End Sub
```
[the remark this answers is not in the document]
Comment (MeSophie) [reply instat/dlgClimograph.vb:308]: This Is solved on PR #8899.

File: instat/dlgClimograph.vb [hunk 6]
```diff
@@ -1,3 +1,363 @@
...
+        ElseIf bRow AndAlso bWrap = False Then
+            clsFacetOperator.AddParameter("right", ".", iPosition:=1)
+        Else
+            clsFacetOperator.RemoveParameterByName("right")
+        End If
+    End Sub
+
+    Private Sub ucr1stFactorReceiver_ControlValueChanged(ucrChangedControl As ucrCore) Handles ucr1stFactorReceiver.ControlValueChanged
+        If Not ucr1stFactorReceiver.IsEmpty Then
+            clsGgwalterliethFunction.AddParameter("station", Chr(34) & ucr1stFactorReceiver.GetVariableNames(False) & Chr(34), iPosition:=1)
```
[the remark this answers is not in the document]
Comment (MeSophie) [reply instat/dlgClimograph.vb:289]: This is solved in PR #8899.

File: instat/dlgClimograph.vb [hunk 7]
```diff
@@ -1,3 +1,363 @@
...
+    End Sub
+
+    Private Sub ucrInput_ControlValueChanged(ucrChangedControl As ucrInputComboBox) Handles ucrInputStation.ControlValueChanged
+        If Not bUpdateComboOptions Then
+            Exit Sub
+        End If
+        Dim strChangedText As String = ucrChangedControl.GetText()
+        If strChangedText <> strNone Then
+            If Not strChangedText = strFacetCol AndAlso Not strChangedText = strFacetRow AndAlso
+                    Not ucrInputStation.Equals(ucrChangedControl) AndAlso ucrInputStation.GetText() = strChangedText Then
```
[the remark this answers is not in the document]
Comment (MeSophie) [reply instat/dlgClimograph.vb:203]: > just for consistency
> 
> ```
> If strChangedText <> strFacetCol AndAlso strChangedText <> strFacetRow AndAlso
>                     ucrInputStation <> ucrChangedControl AndAlso ucrInputStation.GetText() = strChangedText Then
> ```

@N-thony  I obtaind and error on  `ucrInputStation <> ucrChangedControl ` saying `<>` is not define for ucrInputComboBox type.

**Comment by MeSophie** [comment]
I close this because of merge conflicts.

**Pull request closed without merging**